### PR TITLE
handle xen disk labels correctly (xvda-xvdh)

### LINKED ISSUE #1404: Block device assignment should support xen and kvm device names

### PR DESCRIPTION
**Pull request by displague**
This closes #1404

### CODE REVIEW

File: src/linodes/linode/settings/advanced/components/CreateOrEditConfig.js [hunk 1]
```diff
@@ -247,7 +249,7 @@ export default class CreateOrEditConfig extends Component {
           </div>
         </FormGroup>
         <h3 className="sub-header">Block Device Assignment</h3>
-        {AVAILABLE_DISK_SLOTS.map((slot, i) => (
+        {AVAILABLE_DISK_SLOTS[this.props.linode.hypervisor].map((slot, i) => (
```
Comment (eatonphil) [inline src/linodes/linode/settings/advanced/components/CreateOrEditConfig.js]: You already pulled linode out of props in line 118.

File: src/linodes/linode/settings/advanced/components/CreateOrEditConfig.js [hunk 2]
```diff
@@ -403,8 +405,11 @@ CreateOrEditConfig.propTypes = {
 
 CreateOrEditConfig.defaultProps = {
   config: {
-    devices: AVAILABLE_DISK_SLOTS.reduce((disks, slot) => ({ ...disks, [slot]: null }), {}),
-    root_device: '/dev/sda',
+    devices: {
+      ..._.reduce(AVAILABLE_DISK_SLOTS.kvm, (disks, slot) => ({ ...disks, [slot]: null }), {}),
+      ..._.reduce(AVAILABLE_DISK_SLOTS.xen, (disks, slot) => ({ ...disks, [slot]: null }), {}),
```
Comment (eatonphil) [inline src/linodes/linode/settings/advanced/components/CreateOrEditConfig.js]: I don't believe this should be here at all. Just `devices: {}` should work.

**Comment by eatonphil** [comment]
You missed a spot:

screen shot 2017-09-08 at 2 48 13 pm

Also, the default boot device should not default to custom. I think that happend when you changed root_device's default from '/dev/sda' to ''. This is not ideal behavior. Can you make it so it does not default to the Custom boot device selected?

File: src/linodes/linode/settings/advanced/components/CreateOrEditConfig.js (hunk 1 above)
Comment (displague) [reply src/linodes/linode/settings/advanced/components/CreateOrEditConfig.js]: thanks - I pushed over the last commit with this fix.

**Pull request merged**